### PR TITLE
feat(sdk-crash-detection): Store event to dedicated project

### PR DESCRIPTION
**Pull request by philipphofmann**
Stores the detected SDK crash event to a configurable Sentry project.

This PR is based on https://github.com/getsentry/sentry/pull/49528.

Fixes GH-46177

### CODE REVIEW

File: src/sentry/utils/sdk_crashes/sdk_crash_detection.py [hunk 1]
```diff
@@ -13,8 +17,11 @@ class SDKCrashReporter:
     def __init__(self):
         self
 
-    def report(self, event: Event) -> None:
-        return
+    def report(self, event_data: Dict[str, Any], event_project_id: int) -> Event:
+        from sentry.event_manager import EventManager
+
+        manager = EventManager(event_data)
```
Comment (jjbayer) [inline src/sentry/utils/sdk_crashes/sdk_crash_detection.py:23]: Not sure if this is necessary, but better safe than sorry:

```suggestion
        manager = EventManager(event_data)
        manager.normalize()
```

File: src/sentry/utils/sdk_crashes/sdk_crash_detection.py [hunk 2]
```diff
@@ -29,10 +36,9 @@ def __init__(
         self.cocoa_sdk_crash_detector = sdk_crash_detector
         self.event_stripper = event_stripper
 
-    def detect_sdk_crash(self, event: Event) -> None:
-
+    def detect_sdk_crash(self, event: Event) -> Event:
```
Comment (jjbayer) [inline src/sentry/utils/sdk_crashes/sdk_crash_detection.py:39]:
```suggestion
    def detect_sdk_crash(self, event: Event) -> Optional[Event]:
```

If you want to enforce correctness of type hints, add `src/sentry/utils/sdk_crashes` to mypy.ini and run `make backend-typing`.

File: src/sentry/utils/sdk_crashes/sdk_crash_detection.py [hunk 3]
```diff
@@ -44,29 +50,39 @@ def detect_sdk_crash(self, event: Event) -> None:
...
-            sdk_crash_event = self.event_stripper.strip_event_data(event)
+            sdk_crash_event_data = self.event_stripper.strip_event_data(event)
 
             set_path(
-                sdk_crash_event.data, "contexts", "sdk_crash_detection", value={"detected": True}
+                sdk_crash_event_data, "contexts", "sdk_crash_detection", value={"detected": True}
+            )
+
+            if settings.SDK_CRASH_DETECTION_PROJECT_ID is None:
+                return None
```
Comment (jjbayer) [inline src/sentry/utils/sdk_crashes/sdk_crash_detection.py:73]: If `SDK_CRASH_DETECTION_PROJECT_ID` is `None`, there is no benefit in running any of the SDK crash reporting code, or is there? If not, I would move this check to the top-level function in `post_process`.

**Review by jjbayer — CHANGES_REQUESTED** [review]
Looks good to me, but I would move the settings check to an earlier point.

File: src/sentry/utils/sdk_crashes/sdk_crash_detection.py (hunk 2 above)
Comment (philipphofmann) [reply src/sentry/utils/sdk_crashes/sdk_crash_detection.py:36]: I will do that in an extra PR. I created an issue for that https://github.com/getsentry/sentry/issues/49938.

**Review by jjbayer — APPROVED** [review]

**Comment by priscilawebdev** [comment]
I think due to a merge/rebase more people were assigned to review your PR :)

**Pull request merged**